addon/model/model-element.ts

```diff
@@ -336,6 +344,53 @@ export default class ModelElement extends ModelNode implements Cloneable<ModelEl
     }
   }
 
+  getVocab(): string | undefined {
+    if (this.getRdfaPrefixes().has("")) {
+      return this.getRdfaPrefixes().get("");
+    }
+    else {
+      return;
+    }
+  }
+
+  getRdfaPrefixes(): Map<string,string> {
+    // NOTE: we map vocab to an empty string prefix, because this is convenient for passing to children
+    //       it's also conviently how marawa uses it in the RdfaAttributes class
+    const vocab = this.getAttribute("vocab");
+    if (vocab) {
+      return new Map([...this._currentRdfaPrefixes, ["", vocab]]);
+    }
+    else {
+      return new Map([...this._currentRdfaPrefixes]);
+    }
+  }
+
+  updateRdfaPrefixes(prefixes: Map<string,string> = this._currentRdfaPrefixes) {
+    const myPrefixString = this.getAttribute('prefix');
+    if (myPrefixString) {
+      const myPrefixes = parsePrefixString(myPrefixString);
+      this._currentRdfaPrefixes = new Map([...prefixes, ...myPrefixes]);
+    }
+    else {
+      this._currentRdfaPrefixes = prefixes;
+    }
+  }
+
+  setAttribute(key: string, value: string) {
+    super.setAttribute(key, value);
+    if (key === 'prefix') {
+      this.updateRdfaPrefixes();
+    }
+  }
+
+  /*
+   * returns a parsed representation of the rdfa attributes
+   * prefixes are expanded and multivalue attributes are split on space and returned as an array
+   */
+  getRdfaAttributes(): RdfaAttributes {
+    return new RdfaAttributes(this, Object.fromEntries(this.getRdfaPrefixes()));
```

Comment:
open to suggestions on how to best address the linting error here

Reply:
it's simply a version mismatch
our tsconfig is targetting es2017, while this method was only introduced in es2019
if we can afford to target that version, we can just update the tsconfig

Reply:
I don't think es2019 is fully supported by our target browsers yet, but this method is.

Reply:
man writing javascript really is like trying to drive a car while the wheel is still being invented

Ill see what I can do with the tsconfig

Reply:
adding this line to the tsconfig while leaving target unchanged seems to do the trick:
`    "lib": ["ES2017", "ES2019.Object", "DOM", "DOM.Iterable"],`

Reply:
I've found a better table, and according to this one all es2019 are supported atm: https://kangax.github.io/compat-table/es2016plus/ . So I'll go for targetting es2019. The other solution was still giving me issues, even when casting.